Index: llvm-toolchain-snapshot_16~++20220920091612+a7d2409bac7f/clang/lib/Driver/ToolChains/Arch/X86.cpp
===================================================================
--- llvm-toolchain-snapshot_16~++20220920091612+a7d2409bac7f.orig/clang/lib/Driver/ToolChains/Arch/X86.cpp
+++ llvm-toolchain-snapshot_16~++20220920091612+a7d2409bac7f/clang/lib/Driver/ToolChains/Arch/X86.cpp
@@ -114,8 +114,8 @@ std::string x86::getX86TargetCPU(const D
   case llvm::Triple::FreeBSD:
     return "i686";
   default:
-    // Fallback to p4.
-    return "pentium4";
+    // Fallback to i686.
+    return "i686";
   }
 }
 
