Index: llvm-toolchain-snapshot_16~++20221220013513+bb7940e25f6c/llvm/include/llvm/TargetParser/X86TargetParser.def
===================================================================
--- llvm-toolchain-snapshot_16~++20221220013513+bb7940e25f6c.orig/llvm/include/llvm/TargetParser/X86TargetParser.def
+++ llvm-toolchain-snapshot_16~++20221220013513+bb7940e25f6c/llvm/include/llvm/TargetParser/X86TargetParser.def
@@ -249,7 +249,7 @@ CPU_SPECIFIC("pentium_mmx", "pentium-mmx
 CPU_SPECIFIC("pentium_ii", "pentium2", 'E', "+cmov,+mmx")
 CPU_SPECIFIC("pentium_iii", "pentium3", 'H', "+cmov,+mmx,+sse")
 CPU_SPECIFIC_ALIAS("pentium_iii_no_xmm_regs", "pentium3", "pentium_iii")
-CPU_SPECIFIC("pentium_4", "pentium4", 'J', "+cmov,+mmx,+sse,+sse2")
+CPU_SPECIFIC("pentium_4", "pentium4", 'J', "+cmov,+mmx,+sse")
 CPU_SPECIFIC("pentium_m", "pentium-m", 'K', "+cmov,+mmx,+sse,+sse2")
 CPU_SPECIFIC("pentium_4_sse3", "prescott", 'L', "+cmov,+mmx,+sse,+sse2,+sse3")
 CPU_SPECIFIC("core_2_duo_ssse3", "core2", 'M', "+cmov,+mmx,+sse,+sse2,+sse3,+ssse3")
