Index: llvm-toolchain-snapshot_16~++20221126040739+502c246519ec/clang-tools-extra/clangd/index/remote/monitor/Monitor.cpp
===================================================================
--- llvm-toolchain-snapshot_16~++20221126040739+502c246519ec.orig/clang-tools-extra/clangd/index/remote/monitor/Monitor.cpp
+++ llvm-toolchain-snapshot_16~++20221126040739+502c246519ec/clang-tools-extra/clangd/index/remote/monitor/Monitor.cpp
@@ -67,8 +67,8 @@ int main(int argc, char *argv[]) {
       google::protobuf::util::MessageToJsonString(Response, &Output, Options);
   if (!JsonStatus.ok()) {
     clang::clangd::elog("Can not convert response ({0}) to JSON ({1}): {2}\n",
-                        Response.DebugString(), JsonStatus.error_code(),
-                        JsonStatus.error_message().as_string());
+                        Response.DebugString(), (int)JsonStatus.code(),
+                        JsonStatus.message().as_string());
     return -1;
   }
   llvm::outs() << Output;
